### PR TITLE
feat: support alias for source to support stage

### PR DESCRIPTION
**Pull request by JackTan25**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
we need to support alias for source, and to support stage table, and we don't need to support streaming/values/pure table for now, maybe we will do these long after. This pr is enough.
Summary about this PR

- Closes #issue
#12595

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12968)

### CODE REVIEW

**Comment by dantengsky** [comment]
askbend:summary

File: src/query/ast/src/ast/statements/merge_into.rs [hunk 1]
```diff
@@ -91,6 +91,7 @@ pub struct MergeIntoStmt {
     pub database: Option<Identifier>,
     pub table_ident: Identifier,
     pub source: MergeSource,
+    pub alias_source: Option<TableAlias>,
```
Comment (youngsofun) [inline src/query/ast/src/ast/statements/merge_into.rs:94]: why not call it source_alias and target_alias?

**Review by youngsofun — APPROVED** [review]

File: src/query/ast/src/ast/statements/merge_into.rs (hunk 1 above)
Comment (JackTan25) [reply src/query/ast/src/ast/statements/merge_into.rs:94]: well, let me change it.

**Comment by JackTan25** [comment]
image
My local test is successful, but ci failed, it give one more same row. But I think it's non business of merge into. Maybe the satge test is bad. cc @youngsofun 

```
## stage file test
statement ok
drop table if exists test_stage;

statement ok
drop table if exists target_table;

statement ok
create table target_table(a int,b string,c string);

statement ok
insert into target_table values(1,'a_1','b_1'),(2,'a_2','b_2');

query TTT
select * from target_table order by a,b,c;
----
1 a_1 b_1
2 a_2 b_2

statement ok
create table test_stage(a int,b string,c string);

statement ok
insert into test_stage values(1,'a1','b1'),(2,'a2','b2'),(3,'a3','b3');

query TTT
select * from test_stage order by a,b,c;
----
1 a1 b1
2 a2 b2
3 a3 b3

statement ok
drop stage if exists s3;

statement ok
drop stage if exists s4;

statement ok
create stage s3 FILE_FORMAT = (TYPE = CSV);

statement ok
create stage s4 FILE_FORMAT = (TYPE = PARQUET);

statement ok
copy into @s3 from (select a,b,c from test_stage order by a,b,c);

statement ok
copy into @s4 from (select a,b,c from test_stage order by a,b,c);

query TTT
select $1,$2,$3 from @s3;
----
1 a1 b1
2 a2 b2
3 a3 b3

query TTT
select $1,$2,$3 from @s4;
----
1 a1 b1
2 a2 b2
3 a3 b3

## test CSV
statement ok
merge into target_table using (select $1,$2,$3 from @s3) as cdc on cast(cdc.$1 as int) = target_table.a when matched then delete when not matched then insert values(cdc.$1,cdc.$2,cdc.$3);

query TTT
select * from target_table order by a,b,c;
----
3 a3 b3

## test parquet
statement ok
truncate table target_table;

query I
select count(*) from target_table;
----
0

statement ok
insert into target_table values(1,'a_1','b_1'),(2,'a_2','b_2');

query TTT
select * from target_table order by a,b,c;
----
1 a_1 b_1
2 a_2 b_2

statement ok
merge into target_table using (select $1,$2,$3 from @s4) as cdc on cdc.$1 = target_table.a when matched then delete when not matched then insert values(cdc.$1,cdc.$2,cdc.$3);

query TTT
select * from target_table order by a,b,c;
----
3 a3 b3

**Comment by youngsofun** [comment]
try `remove @stage` to clear it  after creating?

**Comment by JackTan25** [comment]
bpaahfaSKu
ci problem, we just load data once, but we can get twice.

**Comment by youngsofun** [comment]
```
statement ok
remove @s5;
```

after create

**Review by dantengsky — APPROVED** [review]

**Pull request merged**